src/pipecat/transports/base_input.py

```diff
@@ -189,7 +201,9 @@ async def _handle_vad(self, audio_frame: InputAudioRawFrame, vad_state: VADState
             frame = None
             if new_vad_state == VADState.SPEAKING:
                 frame = UserStartedSpeakingFrame()
-            elif new_vad_state == VADState.QUIET:
+            # TODO: need to double check if this is the expected behavior
+            # Not triggering the UserStoppedSpeakingFrame if the turn analyser is enabled
+            elif new_vad_state == VADState.QUIET and not self.end_of_turn_analyzer:
```

Comment:
Same here.
And if we choose this approach, I will need to refactor to only trigger the `UserStartedSpeakingFrame` once.